File: questions/question_1.4/problem_1.4.rb
def find_and_replace args
  source = args[:source].split('')
  expr1 = args[:expr1]
  expr2 = args[:expr2]
  new_array = []
  while !source.empty?
    last_element = source.pop
    new_array.unshift(last_element == expr1 ? expr2 : last_element)
  end
  new_array.join('')
end

arguments = {source: 'this is only a test', expr1: ' ', expr2: '%20'}

p find_and_replace arguments

File: strings_and_arrays/questions/question_1.4/problem_1.4.js
var find_and_replace = function(args){
  var source = args.source.split('');
  var expr1 = args.expr1;
  var expr2 = args.expr2;
  var new_expression = [];
  var last_element = ''
  while (source.length > 0){
    last_element = source.pop();
    if (last_element == expr1){ last_element = expr2};
    new_expression.unshift(last_element);
  }
  return new_expression.join('')
}

arguments = {'source': 'this is a test', 'expr1': ' ', 'expr2': '%20'}
console.log(find_and_replace(arguments))